package/gluon-radv-filterd/src/mac.h

```diff
@@ -0,0 +1,8 @@
+#include <stdint.h>
+#include <linux/if_ether.h>
+
+#define F_MAC "%02hhx:%02hhx:%02hhx:%02hhx:%02hhx:%02hhx"
+#define F_MAC_IGN "%*2x:%*2x:%*2x:%*2x:%*2x:%*2x"
+#define F_MAC_VAR(var) var[0], var[1], var[2], var[3], var[4], var[5]
+
+typedef uint8_t macaddr_t[ETH_ALEN];
```

Comment:
How about using a struct for macaddr_t? Most uses are behind F_MAC_VAR anyways, and you could use simple assignments instead of memcpy for the struct.

Reply:
There is already a struct for that - ``struct ether_addr``. From manpage http://man7.org/linux/man-pages/man3/ether_aton.3.html

>       The structure ether_addr is defined in <net/ethernet.h> as:
>
>           struct ether_addr {
>               uint8_t ether_addr_octet[6];
>           }